File: csharp/2.Arrays/CyclicRotation.cs
using System;
// you can also use other imports, for example:
// using System.Collections.Generic;

// you can write to stdout for debugging purposes, e.g.
// Console.WriteLine("this is a debug message");

public class Solution
{
    public int[] solution(int[] A, int K)
    {
        int count = A.Length;
        if (count == 0)
        {
            return A;
        }
        K = K % count;
        Array.Reverse(A, 0, count - K);
        Array.Reverse(A, count - K, K);
        Array.Reverse(A, 0, count);
        return A;
    }
}

File: javascript/2.Arrays/CyclicRotation.js
function solution(A, K) {
  const count = A.length;

  if (count === 0) {
    return A;
  }
  K %= count;
  A.splice(0, count, ...A.slice(count - K, count), ...A.slice(0, count - K));
  return A;
}
